src/components/BooksQuery.tsx

```diff
@@ -1,67 +1,59 @@
-import axios from "axios";
-import { useState } from "react";
-import Card from "@mui/material/Card";
+import { SyntheticEvent, useState } from "react";
 import "./books.css";
-import { CardContent, Typography } from "@mui/material";
-type book = {
-  id: number;
-  title: string;
-  author: string;
-  year: string;
-};
+import { useGoogleBooksApi } from "../hooks/useGoogleApi";
 const BooksQuery = () => {
   const [book, setBooks] = useState("");
-  const [list, setList] = useState([]);
-  const submit = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+  const [{ response, isLoading, isError }, searchBooks] =
+    useGoogleBooksApi(book);
+  let loading = isLoading;
+
+  const submit = async (e: SyntheticEvent) => {
     e.preventDefault();
-    if (!book) {
-      alert("Please enter a book name");
-    } else {
-      getData();
-    }
-    console.log(list);
-  };
-  const getData = () => {
-    axios
-      .get(`https://www.googleapis.com/books/v1/volumes?q=${book}`)
-      .then((data) => {
-        setList(setBooksModel(data.data.items));
-      });
+    loading = true;
+    book ? searchBooks(book) : alert("Please insert book name");
+    loading = false;
   };
 
-  const setBooksModel = (data: any) => {
-    const bookDetails = data.map((book: any) => {
-      return {
-        id: Math.random() * 10,
-        title: book.volumeInfo.title,
-        author: { ...book.volumeInfo.authors },
-        year: book.volumeInfo.publishedDate,
-      };
-    });
-
-    return bookDetails;
-  };
   return (
     <>
       <div className="search-bar">
-        <input
-          value={book}
-          onChange={(e) => {
-            setBooks(e.target.value);
-          }}
-        />
-        <button onClick={submit}>FIND</button>
+        <form onSubmit={submit}>
+          <input
+            type="text"
+            value={book}
+            onChange={(e) => {
+              setBooks(e.target.value);
+            }}
+          />
+          <button type="submit">FIND</button>
+        </form>
       </div>
+      {loading ? <div className="loader"></div> : null}
+      {isError ? <div className="error">Error occured</div> : null}
       <div className="books">
-        {list.map((book: book) => (
-          <div key={book.id} className="card-list">
-            <Card>
-              <CardContent>{book.title}</CardContent>
-              <Typography sx={{ mb: 1.5 }} color="text.secondary">
-                {book.author[0]}
-                <p> pupblished in {book.year}</p>
-              </Typography>
-            </Card>
+        {response?.map((book) => (
+          <div key={book.id} className="books-list">
```

Comment:
Do not just use `div` tags. Try to use semantic tags `p, header, footer, main...etc`